### PR TITLE
Nonce cannot be incremented by depositor

### LINKED ISSUE #85: A combination of small issues
#### Description

- [ ] Do not consume nonce when claiming on behalf https://github.com/sherlock-audit/2024-11-tally/issues/32
- [ ] Simplify calculation https://github.com/sherlock-audit/2024-11-tally/issues/34
- [ ] Remove unused import https://github.com/sherlock-audit/2024-11-tally/issues/40
- [ ] Initialize lastOracleUpdateTime https://github.com/sherlock-audit/2024-11-tally/issues/39

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Remove ununsed import
- Make calculation cleaner
- Initialize lastOracleUpdateTime
- Don't consume nonce unless signature is successful

Closes #85

### CODE REVIEW

File: test/GovernanceStakerOnBehalf.t.sol [hunk 1]
```diff
@@ -1284,7 +1283,7 @@ contract ClaimRewardOnBehalf is GovernanceStakerRewardsTest {
 
     // A user deposits staking tokens
     (_depositAmount, _depositId) =
-      _boundMintAndStake(_depositor, _depositAmount, _delegatee, _claimer);
+      _boundMintAndStake(_depositor, _depositAmount, address(0x01), _claimer);
```
Comment (garyghayrat) [inline test/GovernanceStakerOnBehalf.t.sol:1286]: I don't understand why this change was made?

**Review by garyghayrat — APPROVED** [review]
left a question but otherwise looks good.

File: src/BinaryEligibilityOracleEarningPowerCalculator.sol [hunk 2]
```diff
@@ -119,6 +119,7 @@ contract BinaryEligibilityOracleEarningPowerCalculator is Ownable, IEarningPower
     _setOraclePauseGuardian(_oraclePauseGuardian);
     _setDelegateeScoreEligibilityThreshold(_delegateeScoreEligibilityThreshold);
     _setUpdateEligibilityDelay(_updateEligibilityDelay);
+    lastOracleUpdateTime = block.timestamp;
```
Comment (garyghayrat) [inline src/BinaryEligibilityOracleEarningPowerCalculator.sol:122]: could use a test here.

File: src/GovernanceStaker.sol [hunk 3]
```diff
@@ -761,10 +760,7 @@ abstract contract GovernanceStaker is INotifiableRewardReceiver, Multicall {
     uint256 _depositNewEarningPower,
     uint256 _totalEarningPower
   ) internal pure returns (uint256 _newTotalEarningPower) {
-    if (_depositNewEarningPower >= _depositOldEarningPower) {
```
Comment (wildmolasses) [inline src/GovernanceStaker.sol:764]: was this changed because of a different finding?

File: test/GovernanceStakerOnBehalf.t.sol (hunk 1 above)
Comment (alexkeating) [reply test/GovernanceStakerOnBehalf.t.sol:1286]: Good question, this is to avoid a stack too deep error

File: src/GovernanceStaker.sol (hunk 3 above)
Comment (alexkeating) [reply src/GovernanceStaker.sol:764]: Good question

Comment (alexkeating) [reply src/GovernanceStaker.sol:764]: Yes, it was do you think we should keep it the same?
